### ISSUE TITLE
Very slow import: from megatron.core import parallel_state

### ISSUE THREAD

**Issue by basilevh** [nvidia-cosmos/cosmos-predict2#125]
Hello,

I was trying to figure why any training job takes such a long time to start, and it turns out that in this file:
https://github.com/nvidia-cosmos/cosmos-predict2/blob/main/imaginaire/utils/distributed.py
the line:
`from megatron.core import parallel_state`
usually takes between 2 and 6 minutes, which significantly slows down debugging workflows.

However, in that file:
https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/parallel_state.py
I can't spot much besides a torch import (which should be a NOP since it was already imported), and a bunch of global variables that are assigned None.

Do you have any idea why this slowdown exists, and whether it could be mitigated?

Many thanks in advance!

**Comment by fsergiy** [comment]
HI @basilevh 
I don't think we can immediate root-cause, few answers are required from you:
1) Are you running in Conda or Docker env for this training? Please provide your versions of CUDA, Pytorch. 
2) what is your training setup - multi-node/multi-GPU? which ones? Any NCCL logs during the run? 
3) Have you tried with CPU only training? just wonder if the warm-up time is faster.

**Comment by basilevh** [comment]
Whoops sorry!  I missed your reply @fsergiy 

1. Inside docker, pytorch version `2.7.0a0+79aa17489c.nv25.04`, CUDA version 12.9 according to `nvidia-smi` but 12.8 according to `pip list | grep cuda`
2. A lot before and after the slow import, but none at all during, even with `NCCL_DEBUG_SUBSYS=ALL`
3. No, is there an easy way to do this with the current codebase?

Thanks!

**Comment by MaciejBalaNV** [comment]
Hey @basilevh, you're right that just importing `parallel_state` shouldn't impose any significant overhead. The main function we're using from this file is `initialize_model_parallel`, which is more expensive, but in your case it's happening at import time. I don't think we've seen this behavior before. Could you check 2 more things?
1. How long does it take to run `from megatron.core import parallel_state` outside of training, e.g. in Python interactive mode?
2. Does running with just 1 GPU change the overhead in any way?

**Comment by basilevh** [comment]
Hi @MaciejBalaNV ,
1. about 10 seconds
2. yes, it's much faster, same as 1
